Source/Core/Plugins/EventParser/Raygun/RaygunEventParserPlugin.cs

```diff
@@ -0,0 +1,41 @@
+﻿using System;
+using System.Collections.Generic;
+using Exceptionless.Core.Extensions;
+using Exceptionless.Core.Models;
+using Exceptionless.Core.Pipeline;
+
+namespace Exceptionless.Core.Plugins.EventParser.Raygun {
+    [Priority(0)]
+    public class RaygunEventParserPlugin : IEventParserPlugin {
+        private readonly RaygunEventMapper _mapper = new RaygunEventMapper();
+
+        public List<PersistentEvent> ParseEvents(string input, int apiVersion, string userAgent) {
+            if (userAgent != "raygun" && apiVersion != 1)
+                return null;
+            
+            var events = new List<PersistentEvent>();
+            switch (input.GetJsonType()) {
+                case JsonType.Object: {
```

Comment:
Codewise the following works the same: 
```cs
           switch (input.GetJsonType()) {
                case JsonType.Object:
                    RaygunModel model;
                    if (input.TryFromJson(out model)) {
                        PersistentEvent ev = _mapper.Map(model);
                        events.Add(ev);
                    }
                    
                    break;

                case JsonType.Array:
                    // ...
                    break;
            }
```

So basically it's might just be personal preference or related to readability. Haven't seen it done this way before 😄